### ISSUE TITLE
Hash link does not work

### ISSUE THREAD

**Issue by jonnitto** [sitegeist/Sitegeist.Archaeopteryx#5]
If you have an link with an hash (e.g. `#newsletter`) the editor returns following error:

```
Could not find an editor suitable for "#newsletter"
```

And if I try to click `+ Create Link` I got the error:

```
TypeError
can't access property "useResolvedModel", linkType is null
```

**Comment by mficzel** [comment]
We cannot reproduce this. Can you provide more informations

**Comment by jonnitto** [comment]
First, it is not possible the set hashed links:
<img width="690" alt="CleanShot 2022-05-05 at 20 28 57@2x" src="https://user-images.githubusercontent.com/4510166/166993516-5ca369b0-a793-49d8-b220-20a3582a5041.png">

Second, if you set a default value in the yaml file, you'll get this error

**Comment by mficzel** [comment]
Ahh now i got it ... you can workaround that by linking the current document with an anchor. Not exactly the same but close enough for now i assume.

**Comment by jonnitto** [comment]
For me yes, for a customer not…

**Comment by mhsdesign** [comment]
> As there is no anchor option for media files

but this is a matter of definition: https://github.com/sitegeist/Sitegeist.Archaeopteryx/blob/a003751455ccd6759cd6968dea11c08481ee4296/Neos.Ui/core/src/application/LinkTypes/Asset/Asset.tsx#L17 we _could_ allow anchors on assets.

But this is rather a super edgecasy thing ...

About johns problem: Currently its not anticipated that the host is empty but the anchor is set. Instead of treating the href (protocol + host + path + anchor) as one unit, we allow the anchor to be specified as additional options. But this requires the protocol and host to be present.

**Comment by Benjamin-K** [comment]
> But this is rather a super edgecasy thing ...

It maybe is, but it is not disallowed to use anchors in assets and as you see above, there are valid cases where you might need them. 

Example link: [TransnetBW Nachhaltigkeitsbericht (Seite 6)](https://www.transnetbw.de/_Resources/Persistent/2/7/1/f/271fda65a3408b14f0d0554c03abb9a2be85a956/Nachhaltigkeitsupdate%202022.pdf#page=6) 

And as it might brake the editor if you switch from the "old" editor, i'd rather prefer adding it then having a bad editor experience, where a developer is needed to fix the link.

**Comment by lorenzulrich** [comment]
Example:
![image](https://github.com/sitegeist/Sitegeist.Archaeopteryx/assets/1816023/8900cceb-cf83-402c-b0d2-4eb58855babb)

**Comment by lorenzulrich** [comment]
@Benjamin-K I agree that this should be addressed if this package should go into the core - especially if it will work as drop-in replacement. But I think @mficzel would need to agree with a solution to the problem since it is Sitegeist's package.

**Comment by mficzel** [comment]
I think the custom link as defined here https://github.com/sitegeist/Sitegeist.Archaeopteryx/pull/27  is a good solution. Having yet another extra tab "Anchor" or similar will confuse as many editors as it will help.

**Comment by Benjamin-K** [comment]
@mficzel I totally agree, that we should not add another new link editor to this package. The CustomLink solves the original issue described here. But the edge cases mentioned by @mhsdesign are still valid and will break the backend for some users in some places, of they use node or asset links with a hash in the sidebar. So yes, no new tab please! But this editor should keep links working that worked in the old editor.
Shall we create a separate issue for this?

**Comment by lorenzulrich** [comment]
@mficzel As you can see in my screenshot, it would not be an extra tab, but an extra property in the existing tag. This is just a suggestion. But maybe @Benjamin-K and we should open a new issue for this.

**Comment by jonnitto** [comment]
There is also a new package in the wild: https://github.com/PaesslerAG/Paessler.Anchorlink 

I think it would be nice to make Archaeopteryx extendable with own packages…

**Comment by rolandschuetz** [comment]
@lorenzulrich @mficzel I have created a new issue, so the sub-problem mentioned here about existing links and a breaking editor. https://github.com/sitegeist/Sitegeist.Archaeopteryx/issues/59

**Comment by Benjamin-K** [comment]
@mficzel Could the linked PR #60 be merged?

**Comment by Benjamin-K** [comment]
I think, this could be closed now...

**Comment by mhsdesign** [comment]
fyi i think i found a good solution here https://github.com/sitegeist/Sitegeist.Archaeopteryx/pull/89 regarding keeping the anchors of node links. 

This change will probably be part of the 1.6.0 just that you know @lorenzulrich @Benjamin-K @rolandschuetz
